abstractions/java/lib/src/main/java/com/microsoft/kiota/authentication/AllowedHostsValidator.java

```diff
@@ -0,0 +1,51 @@
+package com.microsoft.kiota.authentication;
+
+import java.net.URI;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.annotation.Nonnull;
+
+/** Maintains a list of valid hosts and allows authentication providers to check whether a host is valid before authenticating a request */
+public class AllowedHostsValidator {
+    private HashSet<String> validHosts;
+    /**
+     * Creates a new AllowedHostsValidator.
+     * @param validHosts The list of valid hosts.
+     */
+    public AllowedHostsValidator(@Nonnull final String... allowedHosts) {
+        this.setAllowedHosts(allowedHosts);
+    }
+    
+    /**
+     * Gets the allowed hosts. Read-only.
+     * @return the allowed hosts.
+     */
+    @Nonnull
+    public Set<String> getAllowedHosts() {
+        return Collections.unmodifiableSet(this.validHosts);
+    }
+    /**
+     * Sets the allowed hosts.
+     * @param allowedHosts the allowed hosts.
+     */
+    public void setAllowedHosts(@Nonnull final String... allowedHosts) {
+        validHosts = new HashSet<String>();
+        if(allowedHosts != null) {
+            for (final String host : allowedHosts) {
+                if (host != null && !host.isEmpty())
+                    validHosts.add(host.trim().toLowerCase());
+            }
+        }
+    }
+
+    /**
+     * Checks if the provided host is allowed.
+     * @param uri the uri to check the host for.
+     * @return true if the host is allowed, false otherwise.
+     */
+    public boolean isUrlHostValid(@Nonnull final URI uri) {
+        return validHosts.isEmpty() || validHosts.contains(uri.getHost().trim().toLowerCase());
```

Comment:
Just checking if this is intended but if the list is empty, that means that all hosts are valid. Yes?
It seems a bit odd that the validator sometimes behaves like a blacklist (list is empty so no need block anything) and other times like a whitelist (if url is on the list, you can go through)

Reply:
yep, if the list is empty, allow everything, otherwise allow what's on the list. I'd rather compare that to an open invitation vs registration required invitation to a party :)
The reason I went this way is because I didn't want to enforce weird semantics like "if we have a single element which value is *, then allow everything". I think this will provide a better usability. Thoughts?
A blocklist would be the contrary, everything that's on the list is not allowed, everything else is.

Reply:
Thanks, I see that now and I agree that would be much simpler as having special cases for matching such scenarios as entries in the list would make it more complex. 

I'm okay with this approach so resolving this for now.